src/Integration/Connection/ConnectionWorkflow.cs

```diff
@@ -97,12 +97,12 @@ private ProgressStepDefinition[] CreateConnectionSteps(IProgressController contr
             return new[]
             {
                 new ProgressStepDefinition(connectStepDisplayText, StepAttributes.Indeterminate | StepAttributes.BackgroundThread,
-                    async (cancellationToken, notifications) =>
-                        await this.ConnectionStepAsync(connection, controller, notifications, cancellationToken)),
+                    (cancellationToken, notifications) =>
+                    this.ConnectionStepAsync(connection, controller, notifications, cancellationToken).GetAwaiter().GetResult()),
```

Comment:
Out of interest, was using _async_ ... _await... not working? Was the working getting stuck at this point?

Reply:
The async/await was not working....The second step was executed before the end of the first one. Besides calling `.Wait()` was causing UI thread to be stuck so this wasn't pleasant solution neither.